### PR TITLE
Updated kepfold to automatically search for periods from KEPBLS

### LINKED ISSUE #129: Kepfold should find the period automatically if passed a kepbls'd file
As I understand it the workflow for using kepfold would be

- kepflatten (remove residuals, creates DETSAP flux)
- kepbls (Finds period, adds 4th extension to fits file, adds best period to header file)
- kepfold (Folds on a specified period)

Because after running kepbls there is a new keyword in the fits header for 'period' and 'bjd0', kepfold should automatically search for that and fold on that period if there isn't one specified by the user. This will let the work flow for people searching for planets become more streamlined.

### PR DESCRIPTION
**Pull request by christinahedges**
Now when you run kepfold on a kepbls'd fits file it will use the best fitting period/bjd0.
However, this does mean that instead of running

`kepfold filename 10 100`

You have to run

`kepfold filename -period 10 -bjd0 100`

How does everyone feel about that?

This fixes #129

### CODE REVIEW

File: pyke/kepfold.py [hunk 1]
```diff
@@ -108,6 +108,19 @@ def kepfold(infile, period, bjd0, outfile=None, bindata=False,
 
     if outfile is None:
         outfile = infile.split('.')[0] + "-{}.fits".format(__all__[0])
+    # check if there is period or BJD0 information
+    if np.any([(period is None),(bjd0 is None)]):
```
Comment (mirca) [inline pyke/kepfold.py]: The only problem I see here is that this statement ignores the input from the user whenever only one of them (``period`` or ``bjd0``) is not ``None``.

File: pyke/kepfold.py [hunk 2]
```diff
@@ -108,6 +108,19 @@ def kepfold(infile, period, bjd0, outfile=None, bindata=False,
...
+        # open input file
+        instr = pyfits.open(infile, 'readonly')
+        try:
+            period = instr[3].header['PERIOD']
+            bjd0 = instr[3].header['BJD0']
+            kepmsg.log(logfile,'Using period and BJD0 from KEPBLS',verbose)
+        except:
+            errmsg = ('ERROR -- KEPFOLD: No period information found. Either specify \
```
Comment (mirca) [inline pyke/kepfold.py]: no need to use ``\`` just break the line and add ``'``  e.g.:
```
 errmsg = ('ERROR -- KEPFOLD: No period information found. Either specify'
                   'period and BJD0 or run KEPBLS.')
```

File: pyke/kepfold.py [hunk 3]
```diff
@@ -108,6 +108,19 @@ def kepfold(infile, period, bjd0, outfile=None, bindata=False,
...
+            period and BJD0 or run KEPBLS.'
+                      .format(outfile))
```
Comment (mirca) [inline pyke/kepfold.py]: No need to break this line. Also, ``outfile`` won't be included in the string unless you add ``{}`` somewhere.

**Comment by mirca** [comment]
@christinahedges thanks for working on this! I just left very few minor comments.

File: pyke/kepfold.py (hunk 1 above)
Comment (christinahedges) [reply pyke/kepfold.py]: Roger, will change that

File: pyke/kepfold.py (hunk 2 above)
Comment (christinahedges) [reply pyke/kepfold.py]: Cheers

File: pyke/kepfold.py (hunk 3 above)
Comment (christinahedges) [reply pyke/kepfold.py]: Hahaha I nicked the error message from below, missed that, thanks

**Comment by christinahedges** [comment]
@mirca Thanks for the quick bodied response. I've updated another tiny tweak by having kepfold allow the non-interactive keyword. This will help with k2torials.

k2tutorials.

**Comment by christinahedges** [comment]
@barentsen Could you also review these changes and check you're happy? @mirca and I are happy to merge, I believe.

**Comment by barentsen** [comment]
We are making quite big changes to the API.  I think they are well-justified, but we are going to have to communicate about it every time we do this to avoid surprising or upsetting users. (And to remember what we did.)

@christinahedges @mirca From now on, let's add an entry to `CHANGES.rst` as part of PRs like this? 
 I am going to open a PR to merge such a change log into PyKE in a few moments!

**Comment by barentsen** [comment]
Otherwise this looks good to be merged!

**Comment by mirca** [comment]
Nicely done @christinahedges!

**Comment by barentsen** [comment]
@christinahedges Thank you for adding the entry to CHANGES, you're the best!

**Pull request merged**